ice/cache.py

```diff
@@ -33,7 +33,11 @@ async def async_wrapper(*args, **kwargs):
             cache, key = get_cache_and_key(func, *args, **kwargs)
             if key in cache:
                 return cache[key]
+            cache.close()
+
             result = await func(*args, **kwargs)
+
+            cache, key = get_cache_and_key(func, *args, **kwargs)
```

Comment:
Since this is being called twice, I think it should be split into two functions for each return value. `key = repr(inspect.getcallargs(func, *args, **kwargs))` is potentially expensive.

Reply:
Why is it better to split into 2 fns?

Reply:
To avoid computing the key twice, since it's completely independent from getting the cache.

Reply:
```
def get_cache(func):
    cache_file = (cache_dir / func.__name__).as_posix() + ".sqlite"
    return SQLiteShelf(cache_file, "diskcache")

def get_cache_key(func, *args, **kwargs):
    return repr(inspect.getcallargs(func, *args, **kwargs))
```